atomix/cluster/src/test/java/io/atomix/raft/storage/RaftStorageTest.java

```diff
@@ -54,49 +48,54 @@ public void testCustomConfiguration() throws Exception {
             .withFreeDiskSpace(100)
             .withFlushExplicitly(false)
             .build();
-    assertEquals("foo", storage.prefix());
-    assertEquals(new File(PATH.toFile(), "foo"), storage.directory());
+    assertThat(storage.prefix()).isEqualTo("foo");
+    assertThat(storage.directory()).isEqualTo(new File(PATH.toFile(), "foo"));
   }
 
   @Test
-  public void testStorageLock() throws Exception {
+  public void canAcquireLockOnEmptyDirectory() {
+    // given empty directory in PATH
+
+    // when
     final RaftStorage storage1 =
         RaftStorage.builder().withDirectory(PATH.toFile()).withPrefix("test").build();
 
-    assertTrue(storage1.lock("a"));
+    // then
+    assertThat(storage1.lock("a")).isTrue();
+  }
+
+  @Test
+  public void cannotLockAlreadyLockedDirectory() {
+    // given
+    final RaftStorage storage1 =
+        RaftStorage.builder().withDirectory(PATH.toFile()).withPrefix("test").build();
+    storage1.lock("a");
 
+    // when
     final RaftStorage storage2 =
         RaftStorage.builder().withDirectory(PATH.toFile()).withPrefix("test").build();
 
-    assertFalse(storage2.lock("b"));
+    // then
+    assertThat(storage2.lock("b")).isFalse();
+  }
+
+  @Test
+  public void canAcquireLockOnDirectoryLockedBySameNode() {
```

Comment:
:thought_balloon: Just as an idea regarding another test. If we would make the writing to a file injectable (via dependency injection) we could also fail the writing and write a test whether failing write doesn't lock the storage anymore.

Reply:
:thinking: Ya. May be. I will pass it for now. It would be also good if we can inject a mock filesystem in which we can simulate all kinds of failure.

Reply:
Agree but then we shouldn't longer use Files class :) or at least some wrapper :)

Reply:
I think @npepinpe had look into to similar ideas for testing journal.